lib/utils/bit_list.ex

```diff
@@ -0,0 +1,92 @@
+defmodule LambdaEthereumConsensus.Utils.BitList do
+  @moduledoc """
+    Set of utilities to interact with bit list, represented as {bitstring, len}.
+  """
+  @type t :: {bitstring, integer()}
+  @bits_per_byte 8
+
+  @doc """
+  Creates a new bit_list from bitstring.
+  """
+  @spec new(bitstring) :: t
+  def new(bitstring) when is_bitstring(bitstring) do
+    # Change the byte order from little endian to big endian (reverse bytes).
+    num_bits = bit_size(bitstring)
+    len = length_of_bitlist(bitstring)
+
+    <<pre::integer-little-size(num_bits - 8), last_byte::integer-little-size(@bits_per_byte)>> =
+      bitstring
+
+    decoded = <<pre::integer-size(num_bits - 8), remove_trailing_bit(<<last_byte>>)::bitstring>>
+    {decoded, len}
+  end
+
+  @spec to_bytes(t) :: bitstring
+  def to_bytes({bit_list, len}) do
+    # Change the byte order from big endian to little endian (reverse bytes).
+    r = rem(len, @bits_per_byte)
+    <<pre::integer-size(len - r), post::integer-size(r)>> = bit_list
+
+    <<pre::integer-little-size(len - r), 1::integer-little-size(@bits_per_byte - r),
+      post::integer-little-size(r)>>
+  end
+
+  @doc """
+  True if a single bit is set to 1.
+  Equivalent to bit_list[index] == 1.
+  """
+  @spec set?(t, non_neg_integer) :: boolean
+  def set?({bit_list, _}, index) do
+    skip = bit_size(bit_list) - index - 1
+    match?(<<_::size(skip), 1::size(1), _::bitstring>>, bit_list)
+  end
+
+  @doc """
+  Sets a bit (turns it to 1).
+  Equivalent to bit_list[index] = 1.
+  """
+  @spec set(t, non_neg_integer) :: t
+  def set({bit_list, len}, index) do
+    skip = bit_size(bit_list) - index - 1
+    <<pre::bitstring-size(skip), _::size(1), rest::bitstring>> = bit_list
+    {<<pre::bitstring, 1::1, rest::bitstring>>, len}
```

Comment:
These are all repeated from `BitVector`. Can we extract those to a single library called BitField and call these from both BitVector and BitList?